ql/src/java/org/apache/hadoop/hive/ql/ddl/table/create/like/CreateTableLikeOperation.java

```diff
@@ -157,27 +158,12 @@ private void setTableParameters(Table tbl) throws HiveException {
     }
     // We should copy only those table parameters that are specified in the config.
     SerDeSpec spec = AnnotationUtils.getAnnotation(serdeClass, SerDeSpec.class);
-
-    // for non-native table, property storage_handler should be retained
-    retainer.add(META_TABLE_STORAGE);
-    if (spec != null && spec.schemaProps() != null) {
-      retainer.addAll(Arrays.asList(spec.schemaProps()));
-    }
-
-    String paramsStr = HiveConf.getVar(context.getConf(), HiveConf.ConfVars.DDL_CTL_PARAMETERS_WHITELIST);
-    if (paramsStr != null) {
-      retainer.addAll(Arrays.asList(paramsStr.split(",")));
-    }
-
-    Map<String, String> params = tbl.getParameters();
-    if (!retainer.isEmpty()) {
-      params.keySet().retainAll(retainer);
-    } else {
-      params.clear();
-    }
+    //With Hive-25813, we'll not copy over table properties from the source.
+    //CTLT should should copy column schema but not table properties. It is also consistent with other query engines like mysql, redshift.
+    tbl.getParameters().clear();
```

Comment:
I'm wondering this would introduce incompatibility, such as discovery partitions, compression etc.
There is also a property `hive.ddl.createtablelike.properties.whitelist`, which select the table Properties to copy over in this case.

Reply:
Yeah, that is true. Going forward we don't want to copy over the physical properties like table parameters in CTLT command. This implementation is inline with other SQL engines like Mysql, Redshift e.t.c.
Given that user can still have same tbl properties as that of source table by explicitly specifiying the TBL PROPERTIES(). Eg:  https://github.com/apache/hive/pull/3627/files#diff-fac39ae652bb8a3e42e189a4efec47b5184ebbb4b7db52ce489dfc343844fd91R11

Reply:
cloud we merge this with `createViewLikeTable`?  since both of them only copy over the `logical` properties.

Reply:
I tried that but if do sp we will end up not copying createtabledesc's table properties.

Reply:
In `setTableParameters`, since we do not use `serdeClass` after changes, can we remove it as well?  and `hive.ddl.createtablelike.properties.whitelist` becomes noop, can we mark it `Deprecated` and add some descriptions about the fix?

Reply:
Ack.